src/bluesearch/database/download.py

```diff
@@ -264,3 +306,34 @@ def download_s3_articles(
             output_path = parent_folder / url.split("/")[-1]
             logger.info(f"Downloading {url}")
             bucket.download_file(url, str(output_path), {"RequestPayer": "requester"})
+
+
+def download_gcs_blob(blob: Blob, out_dir: Path, *, flatten: bool = False) -> None:
+    """Download a Google Cloud Storage blob.
+
+    Parameters
+    ----------
+    blob
+        The blob to download.
+    out_dir
+        The output directory.
+    flatten
+        If false (default) then the directory structure encoded in the blob
+        will be recreated, otherwise the downloaded file will be placed
+        directly into the output directory. For example, if the blob name
+        is "my_files/subdir/file.bin" and flatten is true then the file
+        will be downloaded to "<output_dir>/file.bin", otherwise it will be
+        placed into "<output_dir>/my_files/subdir/file.bin".
```

Comment:
I am not really aware of the structure of `arxiv`. Are we sure we won't have any conflict in the naming if we allow flattening ? Also, is flattening not going to be an issue to be able to retrace back where the file comes from ? I am mainly thinking out loud here, but keeping a copy of the source as is might be simplify a comparison between the source and our local copy and knows what is downloaded and what is not.

Reply:
We can totally keep the folder structure if this helps us in the future.

I reckoned the file names are already unique by themselves and just based on that went for the simpler flat directory structure. But maybe keeping the directory structure is better?

Reply:
If we are sure the file names are unique, then it shouldn't be hard to trace back the file in the source and make sure something was downloaded or not. So we can keep it that way!